### DISCUSSION TITLE
How to load,and create pickled object

### DISCUSSION THREAD

**Discussion by cnrgbot** [nengo/nengo-zoo#454] [General Discussion]
> _Archived from the Nengo Discourse forum._ Originally posted by **assadRasheed** on 2018-02-09. (Forum topic #473, post #1 / id 1306)

i am trying to learn python and machine learning. i am trying to implement and run the [hunse](https://github.com/nengo/nengo_extras/blob/master/examples/cuda_convnet/cifar10_spiking_cnn.py) repository .During this i found a file **cifar10-lif-1628.pkl** which is a pickled object.would like to know that how one can find what is this file how to make your own file like this.

    data_mean = X_train.mean(axis=0)
    X_train -= data_mean
    X_test -= data_mean

    # retrieve from https://figshare.com/s/49741f9e2d0d29f68871
    cc_model = load_model_pickle('cifar10-lif-1628.pkl')

    # --- Run model in Nengo
    presentation_time = 0.2

**Comment by cnrgbot** [comment]
> _Archived from the Nengo Discourse forum._ Originally posted by **Eric** on 2018-02-11. (Forum topic #473, post #3 / id 1309)

That file contains the weights. To make your own, you'll have to train a network, using something like [cuda-convnet2](https://github.com/hunse/cuda-convnet2) or keras (I've just started playing around with Keras more, and I hope to add some spiking CIFAR-10 examples in the near future).

**Reply by cnrgbot** [reply]
> _Archived from the Nengo Discourse forum._ Originally posted by **assadRasheed** on 2018-02-12. (Forum topic #473, post #4 / id 1310)

can i use these weights for other image datasets?

**Reply by cnrgbot** [reply]
> _Archived from the Nengo Discourse forum._ Originally posted by **assadRasheed** on 2018-09-12. (Forum topic #473, post #7 / id 1804)

running build.sh  from [hunse](https://github.com/nengo/nengo_extras/blob/master/examples/cuda_convnet/cifar10_spiking_cnn.py) shows the following error trace on  cuda-6.0 with nvidia 340.24 driver version. need your suggestion about this error.


    --- util
    mkdir -p ./bin//src
    g++  -O3 -c -fPIC   -DNUMPY_INTERFACE -I./include -I/usr/include/python2.7 -I/usr/lib/python2.7/dist-packages/numpy/core/include/numpy/ src/matrix.cpp -o ./bin//src/matrix.o
    In file included from /usr/include/python2.7/Python.h:8:0,
                     from src/../include/matrix.h:22,
                     from src/matrix.cpp:17:
    /usr/include/python2.7/pyconfig.h:1161:0: warning: "_POSIX_C_SOURCE" redefined [enabled by default]
    /usr/include/features.h:164:0: note: this is the location of the previous definition
    /usr/include/python2.7/pyconfig.h:1183:0: warning: "_XOPEN_SOURCE" redefined [enabled by default]
    /usr/include/features.h:166:0: note: this is the location of the previous definition
    cd ./bin/ && g++  -O3   -DNUMPY_INTERFACE -shared -Wl,-no-undefined -o libutilpy.so src/matrix.o -L/usr/lib/atlas-base -latlas -lcblas -lpython2.7
    ln -sf ./bin//libutilpy.so .
    --- nvmatrix
    mkdir -p ./bin/release
    mkdir -p ./obj/release/src
    --- cudaconv3
    --- cudaconvnet
    mkdir -p ./bin/release
    mkdir -p ./obj/release/src
    g++ -m64 -O3 -fPIC -o bin/release/_ConvNet.so obj/release/./src/jpeg.cpp.o obj/release/./src/layer_kernels.cu.o obj/release/./src/convnet.cu.o obj/release/./src/layer.cu.o obj/release/./src/copypipeline.cu.o obj/release/./src/data.cu.o obj/release/./src/streambroadcast.cu.o obj/release/./src/gradreducer.cu.o obj/release/./src/util.cu.o obj/release/./src/lr.cu.o obj/release/./src/neuron.cu.o obj/release/./src/weights.cu.o obj/release/./src/actbroadcaster.cu.o obj/release/./src/reducepipeline.cu.o obj/release/./src/worker.cu.o obj/release/./src/memorysource.cu.o obj/release/./src/pyconvnet.cu.o obj/release/./src/cost.cu.o -L/usr/local/cuda/lib64 -lcudart -lpthread -ljpeg -lpython2.7 -L../util -lutilpy -L../nvmatrix -lnvmatrix -L../cudaconv3 -lcudaconv -lcublas -Wl,-rpath=./util -Wl,-rpath=./nvmatrix -Wl,-rpath=./cudaconv3 -shared 
    /usr/bin/ld: cannot find -lcudaconv
    collect2: ld returned 1 exit status
    make: *** [bin/release/_ConvNet.so] Error 1
    --- make-data/pyext
    build.sh: 61: cd: can't cd to make

**Comment by cnrgbot** [comment]
> _Archived from the Nengo Discourse forum._ Originally posted by **Eric** on 2018-02-12. (Forum topic #473, post #5 / id 1311)

You can, but they might not work that well. This is known as "transfer learning", and is a pretty big sub-field of neural networks. I don't know a whole lot about it, but it would certainly depend on what dataset you've trained on and what you're transferring to.

If you wanted to transfer to something like MNIST from CIFAR-10, that might work ok. You would take your CIFAR-10 weights and re-train the final layer (the classifier layer) on MNIST, so that the network knows the MNIST categories (digits).

**Comment by cnrgbot** [comment]
> _Archived from the Nengo Discourse forum._ Originally posted by **assadRasheed** on 2018-03-14. (Forum topic #473, post #6 / id 1353)

i am trying to run MNIST example in nengo but got  error which is given in this  code [snippet](https://forum.nengo.ai/t/cnn-spiking-using-nengo-extras/486)
suggest me how to rectify this error.
